src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -62,12 +63,14 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.Optional;
 import java.util.function.Function;
 
 /** A build rule which compiles one or more Swift sources into a Swift module. */
-public class SwiftCompile extends AbstractBuildRuleWithDeclaredAndExtraDeps {
+public class SwiftCompile extends AbstractBuildRuleWithDeclaredAndExtraDeps implements
+  SupportsInputBasedRuleKey {
```

Comment:
Asking to learn: how is this protocol is used in this PR?

Reply:
same here

Reply:
Will update PR description with more details.